scripts/release.sh

```diff
@@ -133,6 +133,25 @@ git push "${REPOSITORY}" "${RELEASE_TAG}" ${GIT_DRYRUN:-}
 echo "Deploying release to maven central"
 mvn deploy -Prelease -DskipTests=true -DreleaseSigningKey="${GPG_KEY}" -DprocessAllModules=true ${MVN_DEPLOY_DRYRUN:-}
 
+echo "Creating binary distribution"
+mvn package -Pdist -DskipTests=true
+
+if ! command -v gh &> /dev/null
+then
+    echo "gh command could not be found. Please create a release by hand https://github.com/kroxylicious/kroxylicious/releases/new"
+else
+  echo "Creating GitHub release with binary distribution"
+  ORIGINAL_GH_DEFAULT_REPO=$(gh repo set-default -v | (grep -v 'no default repository' || true))
+  gh repo set-default $(git remote get-url ${REPOSITORY})
+  # create GitHub release via CLI https://cli.github.com/manual/gh_release_create
+  if [[ "${DRY_RUN:-false}" == true ]]; then
+    gh release create "${RELEASE_TAG}" ./kroxylicious-*/target/kroxylicious-*-bin.* --title "${RELEASE_TAG}" --notes-file "CHANGELOG.md" --draft
```

Comment:
does this work if the RELEASE_TAG isn't pushed? I'm guessing `git push "${REPOSITORY}" "${RELEASE_TAG}" ${GIT_DRYRUN:-}` wouldn't have pushed it up.

Reply:
[From the `gh release create` docs:](https://cli.github.com/manual/gh_release_create)

> If a matching git tag does not yet exist, one will automatically get created from the latest state of the default branch.

So it should be fine if there isn't a release tag already in `$REPOSITORY`. However, there should always be a tag, because the git commands on lines 129-131 handle this before the release is created:
```shell
git tag -f "${RELEASE_TAG}"

git push "${REPOSITORY}" "${RELEASE_TAG}" ${GIT_DRYRUN:-}
```

Reply:
I'm not against this idea, but I had a thought to experiment with `jreleaser` for  changelog and release note generation. It'd be nice not to have to remember to update the CHANGELOG manually.

https://github.com/ozangunalp/kafka-native/blob/61fe6a3f4b95cd7f404a91d63323efbac5451a02/pom.xml#L409

Reply:
@k-wall I agree that updating the changelog manually seems a bit of a hassle, especially if there's an easier way of generating it.

I hadn't heard of `jreleaser`, but it seems like quite a powerful tool that could possibly replace most of the content of `scripts/release.sh`. In this PR, I was just trying to get a binary distribution attached to our releases, so that I can reference that in the quick start guides on the website. Perhaps there's room for another PR where this tool could be properly explored.